components/style/servo/restyle_damage.rs

```diff
@@ -285,6 +276,13 @@ fn compute_damage(old: &ServoComputedValues, new: &ServoComputedValues) -> Servo
         get_inheritedbox.visibility
     ]);
 
+
+    // Paint worklets may depend on custom properties,
+    // so if they have changed we should repaint.
+    if old.get_custom_properties() != new.get_custom_properties() {
+        damage.insert(REPAINT);
```

Comment:
Not sure how much better we could do, rather annoyingly. We could try to be smart about which properties are being monitored by a paint worklet.